File: java/2348_Number_of_Zero_Filled_Subarrays.java
class Solution {
    public long zeroFilledSubarray(int[] nums) {
        long totalSubarrays = 0;
        long currentSubarrays = 0;
        for (int num : nums) {
            if (num == 0) {
                currentSubarrays++;
                totalSubarrays += currentSubarrays;
            } else {
                currentSubarrays = 0;
            }
        }

        return totalSubarrays;
    }
}

File: kotlin/2348_Number_of_Zero-Filled_Subarrays.kt
class Solution {
    fun zeroFilledSubarray(nums: IntArray): Long {
        var totalSubarrays: Long = 0;
        var currentSubarrays: Long = 0;
        for (i in 0 until nums.size) {
            if (nums[i] == 0) {
                currentSubarrays++;
                totalSubarrays += currentSubarrays;
            } else {
                currentSubarrays = 0;
            }
        }
        return totalSubarrays;
    }
}
